dockstore-webservice/src/test/java/io/dockstore/webservice/TokenResourceTest.java

```diff
@@ -0,0 +1,198 @@
+package io.dockstore.webservice;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.services.oauth2.model.Userinfoplus;
+import io.dockstore.webservice.core.Token;
+import io.dockstore.webservice.core.TokenType;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.jdbi.TokenDAO;
+import io.dockstore.webservice.jdbi.UserDAO;
+import io.dockstore.webservice.resources.TokenResource;
+import io.dropwizard.auth.AuthValueFactoryProvider;
+import io.dropwizard.auth.CachingAuthenticator;
+import io.dropwizard.testing.junit.ResourceTestRule;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author gluu
+ * @since 24/07/18
+ */
+public class TokenResourceTest {
+    private static final TokenDAO tokenDAO = mock(TokenDAO.class);
```

Comment:
Seems like a miscommunication here, I meant that any interaction with Google can be mocked. 
I'd prefer not mocking the DAOs since we use a live database in our integration tests anyway.
Sorry